omod/src/main/webapp/fragments/dashboardwidgets/appointments.gsp

```diff
@@ -0,0 +1,29 @@
+<div class="info-section">
+    <div class="info-header">
+        <i class="${app.icon}"></i>
+        <h3>${ ui.message(app.label).toUpperCase() }</h3>
+        <a href="${ "/" + contextPath + "/spa/home/appointments/patient/" +   patientUuid }" class="right">
+            <i class="icon-share-alt edit-action" title="Edit"></i>
+        </a>
+    </div>
+    <div class="info-body">
+        <ul>
+            <% appointments.each { appointment -> %>
+            <li>
+                <div style="display: block; overflow: hidden; padding-right: 5px; padding-bottom: 2px;">
+                    <div style="float: left">
+                        ${ui.formatDatetimePretty(appointment.startDateTime)}
+                        <br/>
+                        <strong>
+                            ${ appointment.service?.name }
```

Comment:
Unfortunately, AppointmentServiceDefinition is OpenmrsData, not OpenmrsMetadata, so our standard ${ui.format(appointment.service} pattrn doesn't work, though assumedly it would be trivial to add similar support for OpenmrsData to the UI Formatter if we think that is "correct".

Reply:
I don't think that would work, because OpenmrsData does not have a `name` property, so you couldn't do anything with name.  You'd have to use reflection to support anything with a "name" property, that might work.  But I think this is fine as-is.  If we need this to be localized, we can figure that out and code that in outside of the ui.format tag.